Source/DafnyLanguageServer/Workspace/TextDocumentLoader.cs

```diff
@@ -60,73 +59,108 @@ ILogger<ITextDocumentLoader> logger
       return CreateDocumentWithEmptySymbolTable(compilation, ImmutableDictionary<Uri, IReadOnlyList<Diagnostic>>.Empty);
     }
 
-    public async Task<CompilationAfterParsing> LoadAsync(DafnyOptions options, Compilation compilation,
-      CancellationToken cancellationToken) {
+    public async Task<CompilationAfterParsing> ParseAsync(DafnyOptions options, Compilation compilation,
```

Comment:
Finally a method name that matches what it's actually doing :-)
Also, well done splitting parsing and resolving.